src/vs/workbench/services/telemetry/browser/telemetryService.ts

```diff
@@ -50,6 +52,11 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 				appenders.push(telemetryProvider);
 			}
 			appenders.push(new TelemetryLogAppender(loggerService, environmentService));
+			if (remoteAgentService.getConnection() === null) {
```

Comment:
Should not we always register both? We would like to know about state of remote connection at all times.

Reply:
> Should not we always register both?

Then we would have duplicated events? I haven't tested it but it looks like all events should be send using the backend whenever possible if not then the webappender is used